### PR TITLE
Feature/save card

### PR DESCRIPTION
**Pull request by rem1niscence**
* Add model, test and storage for card

* Add functions to save, load all and delete card

### CODE REVIEW

File: cards/functions/delete-card/main.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	if err != nil {
+		return apigateway.LogAndReturnError(err), nil
+	}
+
+	body, err := url.ParseQuery(request.Body)
+	if err != nil {
+		return apigateway.LogAndReturnError(err), nil
+	}
+
+	if body.Get("card_id") == "" {
```
Comment (pablocampogo) [inline cards/functions/delete-card/main.go:32]: Deletes don´t have body bro, this needs to be a query param

File: cards/functions/delete-card/main.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+		return apigateway.NewErrorResponse(400, ErrMissingCardID), nil
+	}
+
+	err = storage.DeleteCard(ctx, random.GetSHA256WithPrefix(clientModel.ClientIDPrefix, username), body.Get("card_id"))
+	if err != nil {
+		return apigateway.NewErrorResponse(400, err), nil
+	}
+
+	return apigateway.NewJSONResponse(204, nil), nil
```
Comment (pablocampogo) [inline cards/functions/delete-card/main.go:41]: Add here resource deleted successfully to the response

File: cards/functions/retrieve-cards/main.go [hunk 3]
```diff
@@ -0,0 +1,30 @@
+package main
```
Comment (pablocampogo) [inline cards/functions/retrieve-cards/main.go:1]: Name this lambda get cards please for consistency

File: cards/functions/retrieve-cards/main.go [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+	if err != nil {
+		return apigateway.LogAndReturnError(err), nil
+	}
+
+	cards, err := storage.LoadUserCards(ctx, random.GetSHA256WithPrefix(clientModel.ClientIDPrefix, username))
+	if err != nil {
+		return apigateway.LogAndReturnError(err), nil
+	}
+
+	return apigateway.NewJSONResponse(201, cards), nil
```
Comment (pablocampogo) [inline cards/functions/retrieve-cards/main.go:25]: This is a 200

File: cards/functions/delete-card/main.go [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+	body, err := url.ParseQuery(request.Body)
+	if err != nil {
+		return apigateway.LogAndReturnError(err), nil
+	}
+
+	if body.Get("card_id") == "" {
+		return apigateway.NewErrorResponse(400, ErrMissingCardID), nil
+	}
+
+	err = storage.DeleteCard(ctx, random.GetSHA256WithPrefix(clientModel.ClientIDPrefix, username), body.Get("card_id"))
```
Comment (pablocampogo) [inline cards/functions/delete-card/main.go:36]: You should validate the not found to return a 404

File: cards/functions/delete-card/main.go [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+	if err != nil {
+		return apigateway.NewErrorResponse(400, err), nil
```
Comment (pablocampogo) [inline cards/functions/delete-card/main.go:38]: And all the other possible errors from the database are 500 right?

File: cards/functions/retrieve-cards/main.go [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+	"github.com/aws/aws-lambda-go/lambda"
+)
+
+func apiGatewayHandler(ctx context.Context, request events.APIGatewayProxyRequest) (*events.APIGatewayProxyResponse, error) {
+	username, err := apigateway.GetUsername(request)
+	if err != nil {
+		return apigateway.LogAndReturnError(err), nil
+	}
+
+	cards, err := storage.LoadUserCards(ctx, random.GetSHA256WithPrefix(clientModel.ClientIDPrefix, username))
```
Comment (pablocampogo) [inline cards/functions/retrieve-cards/main.go:20]: Add the 404 for not found

**Review by pablocampogo — CHANGES_REQUESTED** [review]
Solve comments

**Comment by pablocampogo** [comment]
Also you need to associate a trip with a card ID and make it mandatory, and also see how you will mange de bank accounts of the partners?

File: cards/functions/retrieve-cards/main.go (hunk 7 above)
Comment (rem1niscence) [reply cards/functions/retrieve-cards/main.go:20]: This is not looking to retrieve a specific card, so I think an empty list when there are no cards saved is a more appropriate response as I does right now, what do you think?

**Comment by rem1niscence** [comment]
> Also you need to associate a trip with a card ID and make it mandatory, and also see how you will mange de bank accounts of the partners?

The last statement will be done on another PR

**Review by pablocampogo — APPROVED** [review]

**Pull request merged**